tests/unit/panther_analysis_tool/test_main.py

```diff
@@ -167,6 +169,47 @@ def test_unknown_exception(self):
         assert_equal(return_code, 1)
         assert_equal(len(invalid_specs), 4)
 
+    def test_with_invalid_mocks(self):
+        args = pat.setup_parser().parse_args('test --path tests/fixtures --filter Severity=Critical RuleID=Example.Rule.Invalid.Mock'.split())
+        args.filter = pat.parse_filter(args.filter)
+        return_code, invalid_specs = pat.test_analysis(args)
+        assert_equal(return_code, 1)
+        assert_equal(len(invalid_specs), 4)
+
+    def test_validate_outputs(self):
+        example_valid_outputs = [
+            ("dedup", "example title"),
+            ("title", "example title"),
+            ("description", "example description"),
+            ("reference", "example reference"),
+            ("severity", "CRITICAL"),
+            ("runbook", "example runbook"),
+            ("destinations", ["example destination"]),
+            ("destinations", []),
+        ]
+        example_invalid_outputs = [
+            ("dedup", None),
+            ("title", None),
+            ("description", None),
+            ("reference", None),
+            ("severity", "CRITICAL-ISH"),
+            ("severity", None),
+            ("runbook", None),
+            ("destinations", ""),
+            ("destinations", ["", None]),
+        ]
+        invalid = False
+        for valid_invalid_outputs in [example_valid_outputs, example_invalid_outputs]:
+            for each_example in valid_invalid_outputs:
+                result = validate_outputs(each_example[0], each_example[1])
+                if invalid:
+                    assert_false(result[0])
+                    assert_false(result[1] == each_example[1])
+                else:
+                    assert_true(result[0])
+                    assert_equal(result[1], each_example[1])
+            invalid = True
```

Comment:
personal preference: I would take the outer `for` loop out and make two tests - just for clarity on what is actually being tested in the asserts.  Definitely not necessary, though.